src/Compilers/CSharp/Portable/Parser/Lexer.cs

```diff
@@ -1980,7 +1980,31 @@ private void LexSyntaxTrivia(bool afterFirstToken, bool isTrailing, ref SyntaxLi
                     case '#':
                         if (_allowPreprocessorDirectives)
                         {
-                            this.LexDirectiveAndExcludedTrivia(afterFirstToken, isTrailing || !onlyWhitespaceOnLine, ref triviaList);
+                            if (isTrailing || !onlyWhitespaceOnLine)
+                            {
+                                // Directives cannot be in trailing trivia.
+                                // We parse the directive (ignoring its effects like disabled text or defined symbols).
+                                // We extract the text corresponding to the parsed directive
+                                // and add it as a BadToken in SkippedTokensTrivia.
+
+                                var savePosition = TextWindow.Position;
+
+                                this.ParseDirective(isActive: false, endIsActive: false, isFollowingToken);
```

Comment:
>isFollowingToken [](http://example.com/codeflow?start=89&length=16)

I think this value is not consistent with the way we were parsing directive before. And it doesn't match what was suggested in the earlier comment.